cmd/quarterdeck/main.go

```diff
@@ -147,6 +165,45 @@ func serve(c *cli.Context) (err error) {
 // Utility Commands
 //===========================================================================
 
+func listOrgs(c *cli.Context) (err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// If csv path is specified, open the file for writing, otherwise use stdout
+	var w rows.Writer
+	if path := c.String("csv"); path != "" {
+		var f *os.File
+		if f, err = os.Create(path); err != nil {
+			return cli.Exit(err, 1)
+		}
+		defer f.Close()
+
+		w = csv.NewWriter(f)
+		defer w.(*csv.Writer).Flush()
+	} else {
+		w = rows.NewTabRowWriter(tabwriter.NewWriter(os.Stdout, 1, 0, 4, ' ', 0))
+		defer w.(*rows.TabRowWriter).Flush()
+	}
+
+	// Write the header
+	w.Write([]string{"ID", "Name", "Domain", "Projects", "Created", "Modified"})
+
+	// Paginate over all organizations in the database, writing rows.
+	cursor := pagination.New("", "", 100)
+	for cursor != nil {
+		var orgs []*models.Organization
+		if orgs, cursor, err = models.ListAllOrgs(ctx, cursor); err != nil {
+			return cli.Exit(err, 1)
+		}
+
+		for _, org := range orgs {
+			w.Write([]string{org.ID.String(), org.Name, org.Domain, fmt.Sprintf("%d", org.ProjectCount()), org.Created, org.Modified})
+		}
+	}
+
+	return nil
+}
```

Comment:
I simplified this a bit because the `tabwriter` and `csv` package incompatibility has frustrated me for a while …